docs/maui/layouts/UniformItemsLayout.md

```diff
@@ -0,0 +1,113 @@
+---
+title: UniformItemsLayout - .NET MAUI Community Toolkit
+author: VladislavAntonyuk
+description: The UniformItemsLayout is a layout where all rows and columns have the same size.
+ms.date: 04/12/2022
+---
+
+# UniformItemsLayout
+
+[!INCLUDE [docs under construction](../includes/preview-note.md)]
+
+The `UniformItemsLayout` is a layout where all rows and columns have the same size.
+
+## Building an UniformItemsLayout
+
+An `UniformItemsLayout` can be created in XAML or C#:
+
+### XAML
+
+```xml
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+             x:Class="MyProject.MyContentPage">
+
+    <toolkit:UniformItemsLayout>
+        <BoxView BackgroundColor="Blue" HeightRequest="25" WidthRequest="25"/>
+        <BoxView BackgroundColor="Yellow" HeightRequest="25" WidthRequest="25"/>
+        <BoxView BackgroundColor="Red" HeightRequest="25" WidthRequest="25"/>
+        <BoxView BackgroundColor="Black" HeightRequest="25" WidthRequest="25"/>
+    </toolkit:UniformItemsLayout>
+    
+</ContentPage>
+```
+
+### C#
+
+```csharp
+using CommunityToolkit.Maui.Views;
+
+var page = new ContentPage
+{
+    Content = new UniformItemsLayout
+    {
+        Children = 
+        {
+            new BoxView { HeightRequest = 25, WidthRequest = 25, BackgroundColor = Colors.Blue },
+            new BoxView { HeightRequest = 25, WidthRequest = 25, BackgroundColor = Colors.Yellow },
+            new BoxView { HeightRequest = 25, WidthRequest = 25, BackgroundColor = Colors.Red },
+            new BoxView { HeightRequest = 25, WidthRequest = 25, BackgroundColor = Colors.Black }
+        }
+    }
+};
+```
+
+## Customizing an UniformItemsLayout
+
+An `UniformItemsLayout` allows to limit the maximum number of columns and rows:
+
+### XAML
+
+```xml
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+             x:Class="MyProject.MyContentPage">
+
+    <toolkit:UniformItemsLayout MaxRows="1" MaxColumns="1">
+        <BoxView BackgroundColor="Blue" HeightRequest="25" WidthRequest="25"/>
+        <BoxView BackgroundColor="Yellow" HeightRequest="25" WidthRequest="25"/>
+        <BoxView BackgroundColor="Red" HeightRequest="25" WidthRequest="25"/>
+        <BoxView BackgroundColor="Black" HeightRequest="25" WidthRequest="25"/>
```

Comment:
I am intrigued by this example, what will render? Blue and Black?

Reply:
It should be only black

Reply:
Damn! For some reason I read it as there being 2 cells in the grid :facepalm:. Perhaps we can add a little note to explain what happens in this scenario?